### PR TITLE
feat: (Core) introduce ControlValueAccessor to SegmentedButton 

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
closes https://github.com/SAP/fundamental-ngx/issues/3671
#### Please provide a brief summary of this pull request.
There is added ControlValueAccessor to SegmentedButton component. Which bases on buttons rendered inside.
I added examples with reactive form and basic ngModel
Also all of usages in docs have been changed regarding it.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: apps/docs/src/app/core/component-docs/date-picker/examples/date-picker-complex-i18n-example/date-picker-complex-i18n-example.component.html [hunk 1]
```diff
@@ -2,38 +2,35 @@
...
         label="German"
-        (click)="setLocale('de')"
-        [class]="isSelected('de')"
+        value="fr"
     ></button>
     <button
         fd-button
         label="Bulgarian"
-        (click)="setLocale('bg')"
-        [class]="isSelected('bg')"
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/date-picker/examples/date-picker-complex-i18n-example/date-picker-complex-i18n-example.component.html:28]: `isSelected` function can be removed from typescript as well

**Comment by mikerodonnell89** [comment]
On the "pagination per page" example, I have to double-click the segmented button to show it as selected

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html [hunk 2]
```diff
@@ -12,9 +12,7 @@
 
 <div class="example-controls-set">
     <label fd-form-label>Activate tab:</label>
-    <fd-segmented-button>
-        <button fd-button label="Tab 1" (click)="tab1.open(true)"></button>
-        <button fd-button label="Tab 2" (click)="tab2.open(true)"></button>
-        <button fd-button label="Tab 3" (click)="tab3.open(true)"></button>
-    </fd-segmented-button>
+    <button fd-button label="Tab 1" (click)="tab1.open(true)"></button>
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html:15]: missing `<fd-segmented-button>` element

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]

File: libs/core/src/lib/segmented-button/segmented-button.component.ts [hunk 3]
```diff
@@ -17,10 +39,237 @@ import { ChangeDetectionStrategy, Component, HostBinding, ViewEncapsulation } fr
...
+    /** @hidden */
+    private _setInitialState(): void {
+        this._toggleDisableButtons(this._isDisabled);
+        this._pickButtonsByValues(this._currentValue);
+    }
+
+    /** @hidden */
+    private _listenToButtonChanges(): void {
+        this._buttons.changes.pipe(startWith(1))
+            .subscribe(_ => {
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/segmented-button/segmented-button.component.ts:137]: @Betrozov `ViewChildren`/`ContentChildren` Queries are unsubscribed upon destroy undet the hood

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html (hunk 2 above)
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html:15]: It's not needed there

File: apps/docs/src/app/core/component-docs/segmented-button/examples/segmented-button-form-example/segmented-button-form-example.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    <div>
+        <h4>Disabled Forms</h4>
+        <fd-segmented-button formControlName="disabled">
+            <button fd-button label="Left" value="first"></button>
+            <button fd-button label="Middle" value="second"></button>
+            <button fd-button label="Right" value="third"></button>
+        </fd-segmented-button>
+        <br>
+        <small>value: {{customForm.controls?.disabled.value}} </small>
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/segmented-button/examples/segmented-button-form-example/segmented-button-form-example.component.html]: Added

**Comment by JKMarkowski** [comment]
> On the "pagination per page" example, I have to double-click the segmented button to show it as selected
Hi @mikerodonnell89 I'm not able to fix it, I guess it's because of lack of change detection on template passed to pagination, but this bug is stlil there, when changeDetection is manually forced.

**Comment by JKMarkowski** [comment]
Hi @mikerodonnell89  and @Betrozov thanks for review. Your comments have been addressed

**Review by Betrozov — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html (hunk 2 above)
Comment (mikerodonnell89) [reply apps/docs/src/app/core/component-docs/tabs/examples/tab-selection-example/tab-selection-example.component.html:15]: Apologies, you are correct

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**